c/bobyqa_c.f90

```diff
@@ -31,8 +32,8 @@ subroutine bobyqa_c(cobj_ptr, data_ptr, n, x, f, xl, xu, nf, rhobeg, rhoend, &
 type(C_PTR), intent(in), value :: xl
 type(C_PTR), intent(in), value :: xu
 integer(C_INT), intent(out) :: nf
-real(C_DOUBLE), intent(in), value :: rhobeg
-real(C_DOUBLE), intent(in), value :: rhoend
+type(C_PTR), intent(in), value :: rhobeg
```

Comment:
No, these should appear in the same order in which they appear in the function signature. I think the local variables should appear in the same order as these as well, to keep things consistent. They are not in that same order right now, but I would prefer to make this change in a separate PR since it is unrelated to the main purpose of this PR.

Reply:
"No, these should appear in the same order in which they appear in the function signature."

I am talking about the local variables, which do not appear in the function signature. 

You may mean that the local variables should follow the same order as their "counterparts" in the arguments of the function. However, not all local variables have such a counterpart. Such a rule will soon lead to an ad-hoc order. Let's take the alphabetical order instead.

Reply:
> I am talking about the local variables (rather than the arguments), which do not appear in the function signature.

I suppose the comment was made in the wrong place then, since the location of the comment references the variables that appear in the function signature.

In general cleanup should be done in cleanup PRs and feature additions/bug fixes should be done in feature addition/bug fix PRs. The reason for not mixing the two is that a) when a feature addition is mixed with cleanup, it can be difficult to tell when looking at the diff what lines implement the feature and what lines do cleanup and b) sometimes cleanup, even just rearranging lines, can have unwanted impacts. Sometimes those impacts aren't seen right away, and when the impacts are noticed and traced back to their introduction, it can be difficult to tell if the feature was responsible for the impact or the cleanup. If these are kept in separate PRs then it's easier to tell the difference and subsequent investigations can move more quickly (and their write-ups can be shorter and more quickly understood).

To be clear, I'm not arguing against sorting the local variables alphabetically. In fact I think it's a good idea. However I propose it be done in a separate PR after this one is merged (it could be done before, and this PR could be rebased, but I think doing it afterwards is easier).

Reply:
Doing some minor cleanup is OK, like `in` -> `IN`, but rearranging and adding (many) lines at the same time is one of those things where the diff becomes challenging.